### PR TITLE
rhc-11619: add "rhc_client_id" for supporting cloud_connector

### PR DESCRIPTION
**Pull request by thearifismail**
This PR adds a new field `rhc_client_id` to system profile for supporting cloud_connector.

The corresponding HBI PR is https://github.com/RedHatInsights/insights-host-inventory/pull/812

### CODE REVIEW

File: tests/utils/invalids.py [hunk 1]
```diff
@@ -49,6 +49,8 @@
     {"selinux_current_mode": "random"},
     {"selinux_config_file": "random"},
     {"owner_id": "x"*12},
+    {"client_id": "x"*12},
```
Comment (tahmidefaz) [inline tests/utils/invalids.py:52]: Are you trying to check the maxLength here? Then this should be ` {"client_id": "x"*37}`.

File: tests/utils/invalids.py [hunk 2]
```diff
@@ -49,6 +49,8 @@
...
+    {"client_id": "plxi13y1-99ut-3rdf-bc10-84opf904lfad"},
```
Comment (ryandillinfelton) [inline tests/utils/invalids.py:53]: I'd like some confirmation from @tahmidefaz that it's okay to have two invalid values in the test. It looks to me like it's fine though.

**Review by ryandillinfelton — APPROVED** [review]
LGTM. Just want some confirmation about the invalids from Tahmid when he reviews this.

File: tests/utils/invalids.py (hunk 2 above)
Comment (tahmidefaz) [reply tests/utils/invalids.py:53]: Yeah, the more the better :+1:

File: tests/utils/invalids.py (hunk 1 above)
Comment (ryandillinfelton) [reply tests/utils/invalids.py:52]: I interpreted it as trying to test that the pattern enforces a minimum length. Having both would probably be best though.

Comment (tahmidefaz) [reply tests/utils/invalids.py]: I assumed the second invalid test was checking the pattern enforcement

Comment (thearifismail) [reply tests/utils/invalids.py]: yes because the valid pattern take [0-9a-f]

**Review by CodeHeeler — APPROVED** [review]
LGTM, regex works as expected, thanks!

**Comment by lphiri** [comment]
@thearifismail  - we agreed to use a more specific name here - I think we settled on "rhc_client_id"  to make it clear is specific to "RHC".

**Comment by kruai** [comment]
Here are the System Profile validation results using the past 3 days of data.
Validating against the RedHatInsights/inventory-schemas master spec:
```
me:
	Pass:148
	Fail:0
	
```
Validating against this PR's spec:
```
me:
	Pass:148
	Fail:0
	
```

**Comment by kruai** [comment]
I agree with Lindani; I also have "rhc_client_id" written down as the field name we agreed on.

**Review by kruai — APPROVED** [review]
LGTM :+1:

**Review by tahmidefaz — APPROVED** [review]

**Comment by kruai** [comment]
FYI, here's the [HBI PR](https://github.com/RedHatInsights/insights-host-inventory/pull/812).

**Pull request merged**